proximal/halide/src/user-problem/problem-definition.h

```diff
@@ -0,0 +1,110 @@
+#pragma once
+#include <Halide.h>
+
+#include <utility>
+
+using namespace Halide;
+
+#include "prior_transforms.h"
+#include "problem-config.h"
+#include "problem-interface.h"
+#include "prox_operators.h"
+#include "prox_parameterized.h"
+
+namespace problem_definition {
+
+using proximal::prox::ParameterizedProx;
+
+/** Transform between variables z and u.
+ *
+ * This is a demonstration of how ProxImaL should generate Halide code for the
+ * (L-)ADMM solver. In this example, the end-user defines the TV-regularizaed
+ * denoising problem with non-negative constraints in Proximal language. Then,
+ * the ProxImaL-Gen module exports the following functions automatically.
+ *
+ * Currently (Sept, 2022), ProxImaL has not yet implemented Halide-codegen
+ * feature in ProxImal. Here, we manualy craft the transform in Halide code as
+ * an example.
+ */
+struct Transform {
+    constexpr static auto width = problem_config::output_width;
+    constexpr static auto height = problem_config::output_height;
+    constexpr static auto N = problem_config::psi_size;
+
+    /** Compute dx, dy of a two dimensional image with c number of channels. */
+    FuncTuple<N> forward(const Func& z) {
+        /* Begin code-generation */
+        return {K_grad_mat(z, width, height), z};
+        /* End code-generation */
+    }
+
+    Func adjoint(const FuncTuple<N>& u) {
+        /* Begin code-generation */
+        const Func z0 = KT_grad_mat(u[0], width, height);
+        const auto& z1 = u[1];
+        Func s;
+        s(x, y, c) = z0(x, y, c) + z1(x, y, c);
+        return s;
+        /* End code-generation */
```

Comment:
The forward and backward mapping `K`. Refer to issue #59 for the challenges of exporting C++ code from Proximal.